lib/ansible/plugins/action/synchronize.py

```diff
@@ -49,7 +49,7 @@ def _get_absolute_path(self, path):
         return path
 
     def _host_is_ipv6_address(self, host):
-        return ':' in host
+        return ':' in to_text(host)
```

Comment:
This looks good except that we should use an errors strategy other than replace here.  How about this?
```suggestion
        return ':' in to_text(host, errors='surrogate_or_strict')
```